internal/provider/kubernetes/gateway_test.go

```diff
@@ -289,3 +291,131 @@ func TestGatewaysOfClass(t *testing.T) {
 		})
 	}
 }
+
+func TestAddFinalizer(t *testing.T) {
```

Comment:
In Contour we have some integration tests of `Reconcile` methods that use a fake client and don't require the whole `envtest` machinery -- I'm personally a fan of them since they test the whole set of logic rather than individual bits, but can be run simply without `envtest`. Might be a pattern we want to start using for our controllers here, to do better integration testing. See e.g. https://github.com/projectcontour/contour/blob/main/internal/provisioner/controller/gatewayclass_test.go. 

I imagine we'll punt this past our next milestone though.

Reply:
@skriss thanks for the review and feedback. I created https://github.com/envoyproxy/gateway/issues/394 to track this.